lib/rules/space-infix-ops.js

```diff
@@ -75,26 +116,31 @@ module.exports = {
          * @private
          */
         function report(mainNode, culpritToken) {
+            const operator = culpritToken.value;
+            const must = spacesYes[operator] ? "must" : "must not";
+
             context.report({
                 node: mainNode,
                 loc: culpritToken.loc.start,
-                message: "Infix operators must be spaced.",
+                data: { must, operator },
+                message: "Operator `{{operator}}` {{must}} be surrounded by spaces.",
                 fix(fixer) {
+                    const shouldHaveSpaces = spacesYes[operator];
                     const previousToken = sourceCode.getTokenBefore(culpritToken);
-                    const afterToken = sourceCode.getTokenAfter(culpritToken);
-                    let fixString = "";
+                    const nextToken = sourceCode.getTokenAfter(culpritToken);
 
-                    if (culpritToken.range[0] - previousToken.range[1] === 0) {
-                        fixString = " ";
-                    }
+                    let fixString = sourceCode.getText(previousToken);
 
+                    if (shouldHaveSpaces) {
+                        fixString += " ";
+                    }
                     fixString += culpritToken.value;
-
-                    if (afterToken.range[0] - culpritToken.range[1] === 0) {
+                    if (shouldHaveSpaces) {
                         fixString += " ";
                     }
+                    fixString += sourceCode.getText(nextToken);
 
-                    return fixer.replaceText(culpritToken, fixString);
+                    return fixer.replaceText(mainNode, fixString);
```

Comment:
It looks like [line 283](https://github.com/eslint/eslint/blob/be739d06ec74bba1e37c1493b51ce934751ac78d/lib/rules/no-extra-parens.js#L283) of no-extra-parens checks to see if the first character of an identifier is a valid "identifier part" . For example, combining an identifier with `typeof` could accidentally create a new identifier like `typeoffoo` rather than an expression like `typeof foo`. Then the fixer inserts a space in these cases to prevent this from happening.

(Also, if you end up feeling like you're stuck or not able to implement this, feel free to let us know and we can implement it and push to your branch.)